sematic/db/queries.py

```diff
@@ -61,6 +61,24 @@ def get_artifact(artifact_id: str) -> Artifact:
         return session.query(Artifact).filter(Artifact.id == artifact_id).one()
 
 
+def get_users(user_ids: List[str]) -> List[User]:
+    """
+    Get users from the database.
+
+    Parameters
+    ----------
+    user_ids: List[str]
+        List of IDs to retrieve.
+
+    Returns
+    -------
+    List[User]
+        List of users
+    """
```

Comment:
If `user_ids` is N times the same ID, will it return N times the same row?

Reply:
Correct that it will return correctly, it just may be less efficient. As the DB is going through each row, it may do:

```
for row in rows:
    for id in in_clause:
        if row.id == id:
            results.add(row)
```
(probably parallelizes this and leverages the index, but still is going to require the same number of CPU cycles).

thus the query could take `Nrows * N_ids_in_clause`. If `N_ids_in_clause` is 10 of the same id, then this query takes 10 times as long as it needs to if you had pre-deduplicated the ids. It's possible that the DB is smart enough to perform the up-front deduplication itself. But I don't know, and it's really easy & cheap to just pass in `list(set(user_ids))` instead of `user_ids`. Why not do it?